lib/writer_test.c

```diff
@@ -0,0 +1,56 @@
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "writer.h"
+
+pthread_mutex_t test_idx_lock;
+
+int test_deserialize_logline() {
+  char *buf = "{\"id\": 1, \"event_type\": 1, \"value\": \"hello\"}";
+  struct logline *l = malloc(sizeof(struct logline));
+  int status = deserialize_logline(buf, l);
+  int result = EXIT_SUCCESS;
+  if (status != 0) {
+    result = EXIT_FAILURE;
+  }
+  if (l->id != 1 || l->event_type != 1 || strcmp(l->value, "hello") != 0) {
+    result = EXIT_FAILURE;
+  }
+  free(l);
+  return result;
+};
+
+void *threadtask() {
+  char *val = "test test test";
+  for (int i = 0; i < 10; i++) {
+    record(2, val);
+  }
+}
+
+int test_concurrent_record() {
```

Comment:
Does this test have any failure conditions?  I guess that's why it's not called in `main` yet?